Makefile

```diff
@@ -36,12 +36,25 @@ lint: lint.sub
 push: push.sub
 test: test.sub
 
+docker-all: .image
+	docker run -ti \
+		-v $(PWD):/mine/src/github.com/kubernetes-incubator/service-catalog \
+		buildimage make .dinit all
+
+.dinit: glide.yaml
+	make init
+	echo > .dinit
+
+.image: Dockerfile
```

Comment:
I'd consider more unique names of both the target, and the buildimage.

Thinking through some cases such as building 2 different Git clients (or different forks):

1.  build client 1 - image gets built
2.  build client 2 - (newer) image gets built, say with different Dockerfile setup
3.  build again in client 1, image won't get rebuilt, yielding unexpected build results.

we could store image version in the .image file for more accuracy. We could, furthermore, use that image version (sha, ...) for the docker run command. that way, 2 different clients could use their own .image file with different versions without a need to rebuild the image repeatedly.